### PR TITLE
[Woo POS][Cash & Receipts] Prompt alert when woo core versions are no correct and receipt clicked

### LINKED ISSUE #13047: [Woo POS] [Cash & Receipts] Prompt alert when Woo core versions are no correct and receipt clicked

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #13047

### Description

The PR adds toast message when a user clicks on "receipt" button and when WCCore version is not supported
### Testing information

Two scenarios:
* With WC >=9.5.0
* More -> POS -> Make purchase
* Click on "receipts"
* Notice that the mock screen is open

---

* With WC < 9.5.0
* More -> POS -> Make purchase
* Click on "receipts"
* Notice that the toast is shown

### The tests that have been performed

Above
### Images/gif

https://github.com/user-attachments/assets/bf052d4e-4fba-4365-8967-22ac510ea2d3

https://github.com/user-attachments/assets/cc518eae-3a75-40d5-861c-b108227aeb0a

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModelTest.kt [hunk 1]
```diff
@@ -614,92 +618,118 @@ class WooPosTotalsViewModelTest {
...
 
-        // THEN
-        verify(cardReaderFacade, never()).collectPayment(any())
-    }
+            // THEN
+            verify(cardReaderFacade, never()).collectPayment(any())
+        }
 
     @Test
-    fun `when OnStartReceiptFlowClicked is triggered, then update state to ReceiptSending with empty email`() = runTest {
```
Comment (kidinov) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModelTest.kt:695]: This is a valid test, but for some unclear reason, it fails when all tests are run and passes only for this class. I removed it as the investigation is not worth the effort, in my opinion.

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this. Works as expected :shipit:

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModelTest.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModelTest.kt:695]: I believe you are missing `whenever(isReceiptSendingSupported()).thenReturn(true)` this line here which would make the test pass.

Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModelTest.kt:695]: Then, another test breaks 🫨  It looks like that there is state somewhere in between tests

**Pull request merged**